segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/DurableDataLog.java

```diff
@@ -17,14 +17,16 @@
 
 import io.pravega.common.util.CloseableIterator;
 import io.pravega.common.util.CompositeArrayView;
+import io.pravega.shared.health.HealthConnector;
+
 import java.io.InputStream;
 import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 
 /**
  * Defines a Sequential Log that contains contiguous ranges of bytes.
  */
-public interface DurableDataLog extends AutoCloseable {
+public interface DurableDataLog extends AutoCloseable, HealthConnector {
```

Comment:
Same here, make the implementation class to implement this health stuff instead of forcing this to be part of `DurableDataLog`.